Updated docs pages

ojeda-e opened the pull request:
Changes in this PR fixes #58 

Changes include:

- [x] Updated algorithm page.
- [x] Update Usage page for three cases 
- Membrane-only
- Membrane-protein with posres.
- Membrane-protein with no posres.
- [x] Updated visualization page.

Pending:
- [x] Add real data file to the test suite. 
- [x] Add real plots to the Visualization page.

Question:
I left the usage page as if I were going to use a different test data file for each case. 
- Is it a good idea to add a dataset for each one of the cases I have or better to reuse some MDA tests?

Happy to receive any other suggestions. Thanks!

@lilyminium @orbeckst @IAlibay @fiona-naughton

ojeda-e commented:
Hi @lilyminium @orbeckst. I would appreciate some feedback on the pushed changes.

Thanks!

lilyminium commented on docs/source/pages/Usage.rst:
```diff
@@ -1,23 +1,127 @@
+---------------------------------------------------------------------
+
+For membrane-protein systems where the simulation setup has no position
+restraints on the protein, a trajectory preprocessing by the user is required.
```
Could you explain what needs to happen in the preprocessing?

lilyminium commented on docs/source/pages/Visualization.rst:
```diff
@@ -1,23 +1,118 @@
+                                                    n_y_bins=8, 
+                                                    wrap=True).run()
+
+        curvature_upper_leaflet = MembraneCurvature(universe, 
```
```suggestion
        curvature_lower_leaflet = MembraneCurvature(universe, 
```

lilyminium commented on docs/source/pages/Visualization.rst:
```diff
@@ -1,23 +1,118 @@
-A divergent colormap is suggested. 
+        fig, (ax1, ax2) = plt.subplots(ncols=2)
+        for ax, mc, lf in zip((ax1, ax2), curvatures, leaflets):
+        ax.imshow(mc, interpolation='gaussian', cmap='seismic', origin='lower')
```
```suggestion
        	ax.imshow(mc, interpolation='gaussian', cmap='seismic', origin='lower')
```

lilyminium replied:
Similar indentation for the rest of the lines, I suspect

lilyminium commented on docs/source/pages/Visualization.rst:
```diff
@@ -1,23 +1,118 @@
+        ax.set_title('{} Leaflet'.format(lf))
+        ax.axis('off')
+
+Which will return 
+
+|mean_mb_example|
```
I wonder if you could try installing the ``ipython`` [directive](https://matplotlib.org/sampledoc/ipython_directive.html) and getting that to plot this for you. If you choose to do this (up to you), you'd need to:

* add `ipython` to https://github.com/MDAnalysis/membrane-curvature/blob/main/docs/requirements.yaml 
* add `'IPython.sphinxext.ipython_console_highlighting', 'IPython.sphinxext.ipython_directive'` to `extensions` in `conf.py` (see user guide https://github.com/MDAnalysis/UserGuide/blob/e88eeefd93ff982032c880cd270929c8d861e1b7/doc/source/conf.py#L86)
* Rewrite this to start with `.. ipython::` instead of `::`

ojeda-e replied:
Thanks for showing me this! Looks awesome! :D

lilyminium commented on docs/source/pages/Visualization.rst:
```diff
@@ -1,23 +1,118 @@
+        plt.show()
 
-With the first, each element of the array is plotted as a square in a matrix 
+With imshow_, each element of the array is plotted as a square in a matrix 
```
```suggestion
With :func:`~matplotlib.pyplot.imshow`, each element of the array is plotted as a square in a matrix 
```

Squiggle ~ means only "imshow" is shown.

lilyminium commented on docs/source/pages/Visualization.rst:
```diff
@@ -1,23 +1,118 @@
+2. contourf
+-------------------------------
+
+You can use contour plots using `contourf`_. With this approach, contour lines
```
```suggestion
You can use contour plots using :func:`~matplotlib.pyplot.contourf`. With this approach, contour lines
```

lilyminium commented on docs/source/pages/Visualization.rst:
```diff
@@ -1,23 +1,118 @@
+line connects points with the same curvature values.
+
+When plotting using `plt.contourf`, an extra step is required to perform an interpolation. 
+We suggest using `scipy.ndimage`_ gaussian filter as in::
```
```suggestion
We suggest using the :func:`scipy.ndimage.gaussian_filter` as in::
```

lilyminium commented on docs/source/pages/Visualization.rst:
```diff
@@ -1,23 +1,118 @@
+and filled contours of the obtained two-dimensional data are plotted. A contour
+line connects points with the same curvature values.
+
+When plotting using `plt.contourf`, an extra step is required to perform an interpolation. 
```
```suggestion
When plotting using :func:`~matplotlib.pyplot.contourf`, an extra step is required to perform an interpolation. 
```

lilyminium commented on docs/source/pages/Visualization.rst:
```diff
@@ -1,23 +1,118 @@
 
 .. _Matplotlib: https://matplotlib.org
-.. _Pandas: https://pandas.pydata.org
+
+.. _imshow: https://matplotlib.org/stable/api/_as_gen/matplotlib.pyplot.imshow.html
+
+.. _contourf: https://matplotlib.org/stable/api/_as_gen/matplotlib.pyplot.contourf.html
+
+.. _scipy.ndimage: https://docs.scipy.org/doc/scipy/reference/generated/scipy.ndimage.gaussian_filter.html
```
You should be able to replace all of these with intersphinx.

lilyminium commented on docs/source/pages/Algorithm.rst:
```diff
@@ -1,47 +1,121 @@
+and Gaussian (`K`) curvature are calculated according to their respective equations.
+
+For every frame of the trajectory, the surface derived from the `AtomGroup` is
+calculated and stored in :attr:`MembraneCurvature.results.z_surface`.
```
```suggestion
calculated and stored in :attr:`MembraneCurvature.results.z_surface<membrane_curvature.base.MembraneCurvature.results.z_surface>`.
```

I can't click this; you may need to fully qualify the path. I haven't checked the above code but something like that should link to "membrane_curvature.base.MembraneCurvature.results.z_surface", but only show "MembraneCurvature.results.z_surface" to the reader.

ojeda-e replied:
Bruh... this I couldn't fix :( 
I tried:
- :attr:`MembraneCurvature.results.z_surface<membrane_curvature.base.MembraneCurvature.results.z_surface>`
- :attr:`MembraneCurvature.results.z_surface<MembraneCurvature.results.z_surface>`
- :attr:`~MembraneCurvature.results.average_z_surface<membrane_curvature.base.MembraneCurvature.results.average_z_surface>`

None of them worked. Honestly, I can't identify what's the issue here.

lilyminium replied:
Sorry, I think this was a brainfart on my part -- I think it makes sense that you can't make this clickable because `.z_surface` is not an attribute of MembraneCurvature but of the Results object, if that makes sense. My apologies!

lilyminium reviewed (changes requested):
This looks amazing @ojeda-e! Lovely figures as always. I have some comments, mostly on how you can get sphinx to do the work for you.

Regarding test files, I do think it'd be a good idea to add the additional test files that you mention in the examples somewhere so users can immediately use your library, and you can also use them for tests. However, one option that you could consider (if they're small) is not adding them to `MDAnalysis.tests`, but rather adding them to `membrane_curvature.tests`. (If the files are large, they will be better off in MDAnalysis.data`.

pep8speaks commented:
Hello @ojeda-e! Thanks for updating this PR. We checked the lines you've touched for [PEP 8](https://www.python.org/dev/peps/pep-0008) issues, and found:

* In the file [`docs/conf.py`](https://github.com/MDAnalysis/membrane-curvature/blob/6d6ba52443355a290669ba453aa8c3270b405e1b/docs/conf.py):

> [Line 53:54](https://github.com/MDAnalysis/membrane-curvature/blob/6d6ba52443355a290669ba453aa8c3270b405e1b/docs/conf.py#L53): [W291](https://duckduckgo.com/?q=pep8%20W291) trailing whitespace
> [Line 181:77](https://github.com/MDAnalysis/membrane-curvature/blob/6d6ba52443355a290669ba453aa8c3270b405e1b/docs/conf.py#L181): [W292](https://duckduckgo.com/?q=pep8%20W292) no newline at end of file

ojeda-e commented:
The most recent pushed changes include:

- [x] added `ipython` to the `Visualization` page to generate the plots. (Thanks @lilyminium for the suggestion, it looks amazing!)
- [x] added `intersphinx_mapping` to link external docs.
- [x] general requested changes.

Two points to discuss:
- I couldn't fix the links of `MembraneCurvature.results.avearge_z_surface` and other attributes on the `Algorithm` page. Can I just leave it like that for now? I knwo it isn't perfect, but I tried several options and couldn't make it work as it should.

- I am not sure what to do about the example for the case of protein with no posres. I can't make my systems public at this very moment. Maybe there is a system with such features in tests or MDAnalysisData? I'm open to suggestions :) 

Thanks!

ojeda-e commented:
I think this PR is ready for review. 

- [x] All usage cases are included. 
- [x] In visualization, plots render with iPython. 

Thanks @lilyminium for all the suggestions!
@orbeckst @IAlibay @fiona-naughton I would appreciate your feedback here although I think it's ready to go. :)

lilyminium reviewed (approved):
Thanks for making those changes @ojeda-e!

ojeda-e commented:
Thanks for taking the time to double-check the changes and approve the PR @lilyminium! 
I will merge after fixing the PEP8.
